packages/shared/src/contexts/payment/StoreKit.tsx

```diff
@@ -79,63 +137,29 @@ export const StoreKitSubProvider = ({
   const { user, isValidRegion: isPlusAvailable } = useAuthContext();
   const { logSubscriptionEvent } = usePlusSubscription();
   const { growthbook } = useGrowthBookContext();
-  const productIds = useFeature(featureIAPProducts);
-  const productList = useMemo(() => Object.keys(productIds), [productIds]);
   const logRef = useRef<typeof logSubscriptionEvent>();
   logRef.current = logSubscriptionEvent;
 
+  const { data } = useQuery<ProductPricingMetadata[]>({
+    queryKey: generateQueryKey(RequestKey.PricePreview, user, 'ios', 'plus'),
+    queryFn: () => fetchPricingMetadata(ProductPricingType.Plus),
+    enabled: !!user && !!growthbook?.ready && iOSSupportsPlusPurchase(),
+    staleTime: StaleTime.Default,
+  });
+
   const { data: productOptions } = useQuery({
     queryKey: ['iap-products'],
-    enabled: !!productIds && !!growthbook?.ready && iOSSupportsPlusPurchase(),
+    enabled: !!data?.length && !!growthbook?.ready && iOSSupportsPlusPurchase(),
```

Comment:
Do we need `growthbook?.ready` anymore?

Reply:
It was because we loaded the product ID's from GB, so needed to ensure GB was ready before we fired the query.
At least that's why I added it for StoreKit.

Reply:
Ah, right. Now the GB that is needed to be loaded will be coming from API side. So we should not need it here indeed.